applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_flux_condition.cpp

```diff
@@ -28,57 +29,50 @@ Condition::Pointer UPwNormalFluxCondition<TDim, TNumNodes>::Create(IndexType
-        Geom.IntegrationPoints(this->GetIntegrationMethod());
-    const unsigned int NumGPoints = IntegrationPoints.size();
-    const unsigned int LocalDim   = Geom.LocalSpaceDimension();
+    const GeometryType&                             r_geometry = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& r_integration_points =
+        r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+    const unsigned int NumGPoints = r_integration_points.size();
 
     // Containers of variables at all integration points
-    const Matrix& NContainer = Geom.ShapeFunctionsValues(this->GetIntegrationMethod());
-    GeometryType::JacobiansType JContainer(NumGPoints);
-    for (unsigned int i = 0; i < NumGPoints; ++i)
-        (JContainer[i]).resize(TDim, LocalDim, false);
-    Geom.Jacobian(JContainer, this->GetIntegrationMethod());
+    const Matrix& r_n_container = r_geometry.ShapeFunctionsValues(this->GetIntegrationMethod());
+    GeometryType::JacobiansType j_container(NumGPoints);
+    for (auto& j : j_container)
+        j.resize(TDim, r_geometry.LocalSpaceDimension(), false);
+    r_geometry.Jacobian(j_container, this->GetIntegrationMethod());
 
     // Condition variables
-    array_1d<double, TNumNodes> NormalFluxVector;
-    for (unsigned int i = 0; i < TNumNodes; ++i) {
-        NormalFluxVector[i] = Geom[i].FastGetSolutionStepValue(NORMAL_FLUID_FLUX);
-    }
+    Vector normal_flux_vector(TNumNodes);
+    VariablesUtilities::GetNodalValues(r_geometry, NORMAL_FLUID_FLUX, normal_flux_vector.begin());
+
     NormalFluxVariables Variables;
 
-    // Loop over integration points
-    for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+    for (unsigned int integration_point = 0; integration_point < NumGPoints; ++integration_point) {
         // Compute normal flux
-        Variables.NormalFlux = 0.0;
-        for (unsigned int i = 0; i < TNumNodes; ++i) {
-            Variables.NormalFlux += NContainer(GPoint, i) * NormalFluxVector[i];
-        }
+        Variables.NormalFlux = MathUtils<>::Dot(row(r_n_container, integration_point), normal_flux_vector);
 
         // Obtain Np
-        noalias(Variables.Np) = row(NContainer, GPoint);
+        noalias(Variables.Np) = row(r_n_container, integration_point);
 
         // Compute weighting coefficient for integration
-        Variables.IntegrationCoefficient = ConditionUtilities::CalculateIntegrationCoefficient<TDim, TNumNodes>(
-            JContainer[GPoint], IntegrationPoints[GPoint].Weight());
+        Variables.IntegrationCoefficient = ConditionUtilities::CalculateIntegrationCoefficient(
+            j_container[integration_point], r_integration_points[integration_point].Weight());
 
         // Contributions to the right hand side
-        this->CalculateAndAddRHS(rRightHandSideVector, Variables);
+        GeoElementUtilities::AssemblePBlockVector(
+            rRightHandSideVector, -Variables.NormalFlux * Variables.Np * Variables.IntegrationCoefficient);
```

Comment:
Let's eliminate `Variables.Np`, e.g.:
```suggestion
        GeoElementUtilities::AssemblePBlockVector(
            rRightHandSideVector, -Variables.NormalFlux * row(r_n_container, integration_point) * Variables.IntegrationCoefficient);
```

Reply:
done. This made the Variables definition in the hpp almost redundant it has move to the FIC variant where its last use is.